neon_data_models/models/api/klat/socketio.py

```diff
@@ -0,0 +1,417 @@
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS  BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+# OR PROFITS;  OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE,  EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import uuid
+
+from time import time
+from typing import Optional, Dict, List, Literal, Union, Any
+from datetime import datetime
+from pydantic import Field, model_validator, model_serializer
+
+from neon_data_models.models.api.llm import LLMPersona
+from neon_data_models.enum import CcaiState
+from neon_data_models.models.base import BaseModel
+
+
+class GetSttRequest(BaseModel):
+    cid: str = Field(description="Conversation ID associated with the request")
+    sid: str = Field(
+        description="Client Session ID associated with the request"
+    )
+    lang: str = Field(
+        description="BCP-47 Language code associated with audio",
+        default="en-us",
+    )
+    message_id: str = Field(
+        description="Message (Shout) ID associated with the request"
+    )
+    audio_data: str = Field(
+        description="B64-encoded audio file object to transcribe"
+    )
+
+
+class GetSttResponse(BaseModel):
+    transcript: str = Field(description="Transcribed text")
+    lang: str = Field(
+        description="BCP-47 Language code associated with `transcript`",
+        default="en-us",
+    )
+    message_id: str = Field(
+        description="Message (Shout) ID associated with the request"
+    )
+    sid: str = Field(
+        description="Client Session ID associated with the request"
+    )
+    cid: str = Field(description="Conversation ID associated with the request")
+
+    @model_validator(mode="before")
+    @classmethod
+    def parse_context(cls, values):
+        """
+        Parse out message_id, sid, and cid from context if not handled by the
+        Observer module.
+        """
+        values.setdefault(
+            "message_id", values.get("context", {}).get("message_id")
+        )
+        values.setdefault("sid", values.get("context", {}).get("sid"))
+        values.setdefault("cid", values.get("context", {}).get("cid"))
+        return values
+
+
+class GetTtsRequest(BaseModel):
+    cid: str = Field(description="Conversation ID associated with the request")
+    sid: str = Field(
+        description="Client Session ID associated with the request"
+    )
+    message_id: str = Field(
+        description="Message (Shout) ID associated with the request"
+    )
+    text: str = Field(description="Text to generate audio for")
```

Comment:
This may be optional; it looks like this is retrieved from the database after the request is received